### PR TITLE
Add openSUSE instructions and refactor README

### LINKED ISSUE #2711: Add build instructions for openSUSE
Currently there are no instructions for building on _openSUSE Tumbleweed_ or _Leap_. It would be nice to have a minimum set of packages to build _TIC-80_ and not rely on _Fedora_ build instructions (since it also installs a whole _GNOME_ environment).

_openSUSE_ has multiple OS variants:

- [Tumbleweed](https://get.opensuse.org/tumbleweed/)
- [Slowroll (experimental)](https://en.opensuse.org/Portal:Slowroll)
- [Leap](https://get.opensuse.org/leap/)
- [Leap Micro](https://get.opensuse.org/leapmicro/)
- [MicroOS](https://get.opensuse.org/microos/)

The primary ones are _Tumbleweed_ and _Leap_. They can also work on _ARM_ and _ARM64_.

### PR DESCRIPTION
**Pull request by fentras**
This pull request closes #2711 and also refactors the _README_ file according to these changes:

- Fixed uneven spacing and indentation
- Updated hyphens to em dashes where applicable
- Capitalized titles
- Removed `https://` from link titles
- Updated the headers structure and table of contents
- Updated _Install Instructions_ to include a no-installation method

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -226,34 +227,32 @@ git clone --recursive https://github.com/nesbox/TIC-80 && cd TIC-80/build
...
+Install with [Install Instructions](#install-instructions)
 
 ### Arch
-
 run the following commands in the Terminal
 ```
 sudo pacman -S cmake ruby mesa libglvnd glu
 git clone --recursive https://github.com/nesbox/TIC-80 && cd TIC-80/build
 cmake -DBUILD_WITH_ALL=On ..
-make -j4
```
Comment (nesbox) [inline README.md:239]: I think we need this command to build binaries, or we can use `cmake --build . --parallel` instead

**Review by nesbox — CHANGES_REQUESTED** [review]
Hi, we need `make` command or `cmake --build . --parallel`

**Comment by fentras** [comment]
> Hi, we need `make` command or `cmake --build . --parallel`

Yeah, I removed `make -j4` since the command was already in the *Install Instructions* section but I didn't notice the *Arch* section missing another `cmake` command. Will fix that right now.

**Review by nesbox — APPROVED** [review]
Thank you

**Pull request merged**